src/routes/signup/email-password.ts

```diff
@@ -60,23 +62,63 @@ export const signUpEmailPasswordHandler: RequestHandler<
   const ticketExpiresAt = generateTicketExpiresAt(60 * 60 * 24 * 30); // 30 days
 
   // insert user
-  const user = await insertUser({
-    disabled: ENV.AUTH_DISABLE_NEW_USERS,
-    displayName,
-    avatarUrl: getGravatarUrl(email),
-    email,
-    passwordHash,
-    ticket,
-    ticketExpiresAt,
-    emailVerified: false,
-    locale,
-    defaultRole,
-    roles: {
-      // restructure user roles to be inserted in GraphQL mutation
-      data: allowedRoles.map((role: string) => ({ role })),
-    },
-    metadata,
-  });
+  const { rows } = await postgres.runSqlParsed(
+    `INSERT INTO auth.users (
+        disabled, 
+        display_name, 
+        avatar_url,
+        email,
+        password_hash,
+        ticket,
+        ticket_expires_at,
+        email_verified,
+        locale,
+        default_role,
+        metadata
+      ) VALUES (%L, %L, %L, %L, %L, %L, %L, %L, %L, %L, %L) RETURNING id`,
+    [
+      ENV.AUTH_DISABLE_NEW_USERS,
+      displayName,
+      getGravatarUrl(email),
+      email,
+      passwordHash,
+      ticket,
+      ticketExpiresAt,
+      false,
+      locale,
+      defaultRole,
+      metadata,
+    ]
+  );
+
+  if (!Array.isArray(rows[0])) {
+    return;
+  }
+
+  const insertedUserId = rows[0][0];
+
+  // insert allowed roles
+  for (const role of allowedRoles) {
+    await postgres.runSql(
+      `INSERT INTO auth.user_roles (user_id, role) VALUES (%L, %L)`,
+      [insertedUserId, role]
+    );
+  }
```

Comment:
We're losing inserting a user and roles in a transaction here, compared to GraphQL.